plugins/inputs/zfs/zfs_linux.go

```diff
@@ -93,10 +191,10 @@ func (z *Zfs) Gather(acc telegraf.Accumulator) error {
 		kstatPath = "/proc/spl/kstat/zfs"
 	}
 
-	pools := getPools(kstatPath)
+	pools, err := getPools(kstatPath)
 	tags := getTags(pools)
 
-	if z.PoolMetrics {
+	if z.PoolMetrics && err == nil {
```

Comment:
Shouldn't we simply return on the error?
```golang
	pools, err := getPools(kstatPath)
	if err != nil {
		return fmt.Errorf("cannot get pools: %v", err)
	}
	tags := getTags(pools)

	if z.PoolMetrics {
```

Reply:
I was trying to keep the original behavior of collecting the rest of the metrics even when there may be an error (or the config is not enabled) around collecting pool metrics.

I can change it to error our immediately if you'd prefer.

Reply:
I'm happy with your code if this reproduces the previous behavior. Please keep the code you have and I will mark this as resolved. :-)

Reply:
Sounds good, thanks!